### DISCUSSION TITLE
Send message to specific client

### DISCUSSION THREAD

**Discussion by car1ot** [uNetworking/uWebSockets.js#249] [Q&A] [answered]
How to send a message to a specific client? For example, user refill account with money and when transaction with bank is successful, bank send request to my server payments handler with user ID parameters, amount etc.. I want to notify client of successful transaction.

**Comment by ghost** [comment, marked as answer]
ws.send - you **really** need to look up the basic things before asking

**Comment by hst-m** [comment]
You need to either store references to the `ws` objects in Maps by ID and send with `ws.send()` or you can Subscribe the `ws` to the user ID in the ws open hander with ``ws.subscribe(`user/id/${userID}`)`` and then Publish to that ID with ``app.publish(`user/id/${userID}`)``, this works if they have multiple tabs/devices open etc. Unless you only want to send to that specific `ws` you could use ``ws.subscribe(`ws/id/${ID}`)`` @alexhultman is it better to subscribe/publish to a unique ID or store ws objects by ID in JS and use ws.send()?
